examples/vision/aodnet.py

```diff
@@ -0,0 +1,354 @@
+"""
+Title: An All-in-One Network for Dehazing and Beyond
+Author: [Soumik Rakshit](https://github.com/soumik12345)
+Date created: 2021/04/10
+Last modified: 2021/04/10
+Description: Train a lightweight Deep CNN for image denhazing.
+"""
+"""
+# Introduction
+
+Image Dehazing is an important problem is Computer Vision. It can not only be used to
+enhance photographs clicked in hazy conditions, but also can be used to augment other
+computer vision systems such as Object Detection and Seantic Segmentation as a
+pre-processing step. In this example, we implement
+[AODNet](https://arxiv.org/abs/1707.06543v1), a simple and light-weight Deep CNN model
+that directly generates a clean image from a hazy image. The end-to-end design of AODNet
+makes it easy to embed AOD-Net into other deep models such as Faster-RCNN and YOLO as a
+pre-processing step, making it easier for such model to detect and recognize objects from
+a clean image.
+"""
+
+"""shell
+!nvidia-smi -L
+"""
+
+import os
+import gdown
+import numpy as np
+from PIL import Image
+from glob import glob
+import tensorflow as tf
+import matplotlib.pyplot as plt
+
+"""
+# Downloading Dataset
```

Comment:
Suggestion:

```suggestion
# Load the NYU2 Depth Database
```

(similar to https://keras.io/examples/vision/image_classification_from_scratch/)